src/main/java/org/morehouse/restapp/resources/TodosResource.java

```diff
@@ -94,13 +94,28 @@ public String deleteTodo(@PathVariable long id) {
 
 		Optional<Todo> todoOptional = todoRepository.findById(id);
 
-		if (!todoOptional.isPresent())
+		if (!todoOptional.isPresent()) {
 			return ResponseEntity.notFound().build();
+		}
+		if (!validateUpdate(todoOptional.get(), todo)) {
+			return ResponseEntity.badRequest().build();
+		}
 
 		todo.setId(id);
 		
 		todoRepository.save(todo);
 
 		return ResponseEntity.noContent().build();
 	}
+	
+	boolean validateUpdate(Todo existing, Todo update) {
+		if (existing.getAssignee() == null && update.getAssignee() == null) {
```

Comment:
the fact that you are checking `update.getAssignee()` in both this conditional and in the `else` suggests that you might be able to collapse these.

i believe the following like is functionally equivalent to what you have here:  
`return existing.getAssignee() == null || update.getAssignee() != null`.